src/radix_connect/mobile/client.rs

```diff
@@ -52,6 +58,22 @@ impl RadixConnectMobile {
         // 2. Generate a new Diffie-Hellman key pair
         let wallet_private_key = DiffieHellmanPrivateKey::generate()?;
 
+        let dapp_definitions = self
+            .well_known_client
+            .get_well_known_file(request.origin.clone())
+            .await?;
+        // pass dev mode to decide if we want to use default callback path or fail linking
+        if !dev_mode && dapp_definitions.callback_path.is_none() {
+            return Err(
+                CommonError::RadixConnectMobileDappCallbackPathNotFound {
+                    origin: request.origin,
+                },
+            );
+        }
+        let callback_path = dapp_definitions
+            .callback_path
+            .unwrap_or(String::from("default_callback_path"));
```

Comment:
This is a strong argument against using `String`! `.unwrap_or(String::from("default_callback_path"));` is clearly placeholder values. If you create a `newtype` for this:

```rust
#[derive(Clone, PartialEq, Eq, derive_more::FromStr, derive_more::Display, DeserializeFromStr, SerializeDisplay)]
pub struct RCMCallbackPath(pub(crate) String);
```

(if needed you to make it UniFFI compatible take a look at `BIP39Passphrase`).

And then you impl `Default` for it:

```rust
impl Default for RCMCallbackPath {
    fn default() -> Self {
        err!("Using PLACEHOLDER Default RCM CallbackPath");
        Self("default_callback_path".to_owned())
    }
}
```

and then you can use:

```rust
 let callback_path = dapp_definitions
            .callback_path
            . unwrap_or_default();
```

Reply:
and probably a missing `//FIXME replace this placeholder default value`....

Reply:
so `callback_path` in both `Session` struct and `DappDefinition` which is purely model of json response should use `RCMCallbackPath`?